src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -25,12 +25,29 @@
 	<div class="contact-header-avatar">
 		<div class="contact-header-avatar__wrapper">
 			<div class="contact-header-avatar__background" @click="toggleModal" />
+
 			<div v-if="contact.photo"
+				v-click-outside="closeMenu"
 				:style="{ 'backgroundImage': `url(${contact.photoUrl})` }"
-				class="contact-header-avatar__photo"
-				@click="toggleModal" />
+				class="avatar contact-header-avatar__options contact-avatar-options contact-header-avatar__photo"
+				@click="toggleModal">
+				<input id="contact-avatar-upload"
+					ref="uploadInput"
+					type="file"
+					class="hidden"
+					accept="image/*"
+					@change="processFile">
+			</div>
+
+			<div v-if="contact.photo"
+				class="contact-header-avatar__overlay"
+				@click.stop.prevent="toggleMenu">
+				<div class="overlay-icon icon-picture-force-white" />
+			</div>
```

Comment:
Why not use the menu directly here?
I don't understand the need to have a button that opens the menu?

You can customize the default menu icon if that was the issue you were facing :)
`default-icon="icon-picture-force-white"`

Reply:
It's more that I wanted the button to be visible as soon as you hover over the avatar and to change it's opacity as soon as you hover over to the button itself. I could not figure out how to do that with an action button. Is that possible?

|              | action button | with additional button |
|--------------|---------------|------------------------|
| page view | ![image](https://user-images.githubusercontent.com/34400929/87791969-7769ae00-c843-11ea-9033-d26960029249.png) | ![image](https://user-images.githubusercontent.com/34400929/87792732-9b79bf00-c844-11ea-9f40-60028721ccc5.png) |
| hover avatar | ![image](https://user-images.githubusercontent.com/34400929/87792194-d3cccd80-c843-11ea-9bee-6768c71cf7f1.png) | ![image](https://user-images.githubusercontent.com/34400929/87793161-2b1f6d80-c845-11ea-8b72-0c1bee36fd8f.png) |
| hover button | ![image](https://user-images.githubusercontent.com/34400929/87792378-18f0ff80-c844-11ea-83ef-3cbe319e88e7.png) | ![image](https://user-images.githubusercontent.com/34400929/87793231-45f1e200-c845-11ea-87be-0d51e2ed0ed7.png) |

*(icons by [Vectto](https://www.iconfinder.com/icons/2624896/avengers_captain_captain_america_super_hero_icon), CC BY 3.0)*

Reply:
Sure, you should still be able to manage it like that :)
Currently this si weird as there is this two element superposition :thinking:

Reply:
I have pushed my attempts, but need some help with the detection whether the menu is opened. Is there a way to trigger that toggleMenu when the action button is clicked (without putting a div around it)?

Reply:
Thank you for the hint, the .sync solved the issue!